frontend/public/components/marketplace/marketplace-item-modal.jsx

```diff
@@ -1,60 +1,114 @@
 import * as React from 'react';
+import * as _ from 'lodash-es';
 import * as PropTypes from 'prop-types';
 import {Button, Modal} from 'patternfly-react';
 import {CatalogItemHeader, PropertiesSidePanel, PropertyItem} from 'patternfly-react-extensions';
+
+import {k8sCreate, k8sUpdate} from '../../module/k8s';
+import {CatalogSourceConfigModel} from '../../models/index';
 import {MarkdownView} from '../operator-lifecycle-manager/clusterserviceversion';
 
-const MarketplaceItemModal = (props) => {
-  const { item, show, close /* openSubscribe */ } = props;
+const MARKETPLACE_CSC_NAME = 'marketplaced';
 
-  if (!item) {
-    return null;
+const getPackages = (catalogsourceconfigs) => {
+  const marketplaceCSC = _.filter(catalogsourceconfigs.data, (csc) => {
+    const name = _.get(csc, 'metadata.name', false);
+    return name === MARKETPLACE_CSC_NAME;
+  });
+  if (_.isEmpty(marketplaceCSC)) {
+    return '';
   }
+  return _.get(marketplaceCSC[0], 'spec.packages');
+};
 
-  const { name, iconClass, imgUrl, provider, longDescription, description, version, certifiedLevel, healthIndex, repository, containerImage, createdAt, support } = item;
-  const notAvailable = <span className="properties-side-panel-pf-property-label">N/A</span>;
-  const MarketplaceProperty = ({label, value}) => {
-    return <PropertyItem label={label} value={value || notAvailable} />;
-  };
+class MarketplaceItemModal extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  enable() {
+    const {item, catalogsourceconfigs, close} = this.props;
+    const {name} = item;
+    const previousPackages = getPackages(catalogsourceconfigs);
+    const packageArray = _.map(previousPackages.split(','), pkg => pkg.trim());
+    if (_.includes(packageArray, name)){
+      // TODO: display that catalogsource already exists
+      close();
+      return;
+    }
+    const packages = previousPackages ? `${previousPackages},${name}` : `${name}`;
+    const catalogSourceConfig = {
+      apiVersion: `${CatalogSourceConfigModel.apiGroup}/${CatalogSourceConfigModel.apiVersion}`,
+      kind: `${CatalogSourceConfigModel.kind}`,
+      metadata: {
+        name: `${MARKETPLACE_CSC_NAME}`,
+        namespace: 'openshift-operators',
+      },
+      spec: {
+        targetNamespace: 'openshift-operators',
+        packages: `${packages}`,
+      },
+    };
+    if (!previousPackages) {
+      k8sCreate(CatalogSourceConfigModel, catalogSourceConfig);
+    } else {
+      // NOTE: doesn't work for some reason - don't know why...
+      k8sUpdate(CatalogSourceConfigModel, catalogSourceConfig);
```

Comment:
Have you tried used `k8sCreate` instead? Using `oc create -f` seems to replace/update the `CatalogSourceConfig` when I try it.

Reply:
Excellent! Been banging my head trying to debug this for the last few minutes! Thanks

Reply:
actually I still get 
```
Uncaught (in promise) Error: catalogsourceconfigs.marketplace.redhat.com "marketplaced" already exists
```
when trying to update the existing csc